contracts/wawaka/exchange/contracts/common/IssuerAuthority.h

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "StringArray.h"
+#include "Value.h"
+
+#include "Common.h"
+#include "StateReference.h"
+
+#define ISSUER_AUTHORITY_SCHEMA "{"                        \
+    SCHEMA_KW(authorized_issuer_verifying_key,"") ","           \
+    "\"issuer_state_reference\":" STATE_REFERENCE_SCHEMA ","    \
+    SCHEMA_KW(authorizing_signature,"")                         \
+    "}"
+
+namespace ww
+{
+namespace exchange
+{
+
+    class IssuerAuthority : public ww::value::Structure
```

Comment:
on the scale of inefficiencies, this is a very minor one (there are way too many copies in several places). since the point of using the structure is to provide some (simplistic) enforcement of fields, a better solution would be to make get_value smarter. suggest we just add it to the list of future things to do.